velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -47,12 +47,13 @@ class ReaderBase {
     return fileLength_;
   }
 
-  uint64_t fileNumRows() const {
-    return fileMetaData_->num_rows;
+  const thrift::FileMetaData& thriftFileMetaData() const {
+    return *fileMetaData_;
   }
 
-  const thrift::FileMetaData& fileMetaData() const {
-    return *fileMetaData_;
+  const std::unique_ptr<FileMetaData> fileMetaData() const {
```

Comment:
Since this is just a thin wrapper over `const thrift::FileMetaData*`, how about we just return the object on stack?
```
FileMetaData fileMetaData() const;
```